src/routers/alpha-router/alpha-router.ts

```diff
@@ -630,12 +630,26 @@ export class AlphaRouter
       }
       exchangeRate = swap.trade!.outputAmount.divide(swap.trade!.inputAmount);
 
-      log.info({
-        optimalRatio: optimalRatio.asFraction.toFixed(18),
-        newRatio: newRatio.asFraction.toFixed(18),
-        ratioErrorTolerance: swapAndAddConfig.ratioErrorTolerance.toFixed(18),
-        iterationN: n.toString(),
-      });
+      log.info(
+        {
+          exchangeRate: exchangeRate.asFraction.toFixed(18),
+          optimalRatio: optimalRatio.asFraction.toFixed(18),
+          newRatio: newRatio.asFraction.toFixed(18),
+          inputBalanceUpdated: inputBalanceUpdated.asFraction.toFixed(18),
+          outputBalanceUpdated: outputBalanceUpdated.asFraction.toFixed(18),
+          ratioErrorTolerance: swapAndAddConfig.ratioErrorTolerance.toFixed(18),
+          iterationN: n.toString(),
+        },
+        'QuoteToRatio Iteration Parameters'
+      );
+
+      if (exchangeRate.equalTo(0)) {
+        log.info('exchangeRate to 0');
+        return {
+          status: SwapToRatioStatus.NO_ROUTE_FOUND,
```

Comment:
bc of math, without catching this, we'd return the error on line 562, which is misleading.